### PR TITLE
Common improvements EnvConfig and SiteConfig

### PR DESCRIPTION
**Pull request by Chrico**
SiteConfig // remove spaces.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

- EnvConfig - added new env constant `EnvConfig::TRAINING` with mapping for `training` and `train´ to this constant.
- SiteConfig - removed "Spaces" hosting, which does not exist anymore.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No.

### CODE REVIEW

File: src/EnvConfig.php [hunk 1]
```diff
@@ -28,6 +29,8 @@ class EnvConfig implements SiteConfig
         'preprod' => self::STAGING,
         'pre-prod' => self::STAGING,
         'pre-production' => self::STAGING,
+        'training' => self::TRAINING,
+        'train' => self::TRAINING,
```
Comment (gmazzap) [inline src/EnvConfig.php:33]: This is something very specific to one of our clients (this repo is public so I'm mentioning it, but you know). I don't think it should be in this generic and public package.

Besides that, the value of the array is supposed to be a value returned by `wp_get_environment_type()` so we can map arbitrary env names to one of the 4 supported by WP, and "training" is _not_ supported by VIP.

File: src/EnvConfig.php [hunk 2]
```diff
@@ -38,7 +41,6 @@ class EnvConfig implements SiteConfig
     private const HOSTING_LOCATIONS_CLASS_MAP = [
         self::HOSTING_WPE => WpEngineLocations::class,
         self::HOSTING_VIP => VipLocations::class,
-        self::HOSTING_SPACES => GenericLocations::class,
```
Comment (gmazzap) [inline src/EnvConfig.php:41]: Why removing it? I mean, even if _we_ don't use it maybe someone else will? This is a public repo...

Comment (Chrico) [reply src/EnvConfig.php:41]: Spaces hosting does not exist anymore, they closed the service. :)

**Comment by Chrico** [comment]
I'm going to close this PR, since it is mixing up 2 different topics and 1 of them will not land into it. Thanks for your feedback @gmazzap 👍🏻

**Pull request closed without merging**